ops/testing.py

```diff
@@ -446,18 +475,74 @@ def add_relation_unit(self, relation_id: int, remote_unit_name: str) -> None:
         relation_name = self._backend._relation_names[relation_id]
         # Make sure that the Model reloads the relation_list for this relation_id, as well as
         # reloading the relation data for this unit.
-        if self._model is not None:
-            remote_unit = self._model.get_unit(remote_unit_name)
-            relation = self._model.get_relation(relation_name, relation_id)
-            unit_cache = relation.data.get(remote_unit, None)
-            if unit_cache is not None:
-                unit_cache._invalidate()
-            self._model.relations._invalidate(relation_name)
+        remote_unit = self._model.get_unit(remote_unit_name)
+        relation = self._model.get_relation(relation_name, relation_id)
+        unit_cache = relation.data.get(remote_unit, None)
+        if unit_cache is not None:
+            unit_cache._invalidate()
+        self._model.relations._invalidate(relation_name)
         if self._charm is None or not self._hooks_enabled:
             return
         self._charm.on[relation_name].relation_joined.emit(
             relation, remote_unit.app, remote_unit)
 
+    def remove_relation_unit(self, relation_id: int, remote_unit_name: str) -> None:
+        """Remove a unit from a relation.
+
+        Example::
+
+          rel_id = harness.add_relation('db', 'postgresql')
+          harness.add_relation_unit(rel_id, 'postgresql/0')
+          ...
+          harness.remove_relation_unit(rel_id, 'postgresql/0')
+
+        This will trigger a `relation_departed` event. This would
+        normally be followed by a `relation_changed` event triggered
```

Comment:
Is this actually true? I don't remember seeing a relation-changed as a result of relation-departed. It is true of relation-joined, but I would think departed means the unit is gone and thus we don't have a changed event related to that unit.

Reply:
Thank you @jameinel . Ah. Looks like I wrote that backwards. Consulting [ops docs](https://ops.readthedocs.io/en/latest/#ops.charm.RelationDepartedEvent) again, can I rephrase this as saying "This will normally be proceeded by a relation changed event" ...